_test/template.response.test.php

```diff
@@ -0,0 +1,257 @@
+        $header = $header->nodeValue;
+
+        $expectedHeader = 'FINDOLOGIC XML Export Plugin';
+
+        $legend = $dom->getElementsByTagName('legend')[0];
+        $legend = $legend->nodeValue;
+
+        $expectedLegend = 'Pages without title (1)';
+
+        $this->assertEquals($expectedHeader, $header, 'Expected header should be equal to the translation set in the english translation file.');
+        $this->assertEquals($expectedLegend, $legend, 'Expected legend should be equal to the translation set in the english translation file and should contain one item when one page has no title set.');
+    }
+
+    public function test_response_has_one_element_when_two_pages_has_no_title_set()
+    {
+        $pageHasNoTitle = ['ihavenotitle1', 'ihavenotitle2'];
+        Helper::savePages($pageHasNoTitle);
+
+        ob_start();
+
+        $adminPlugin = new admin_plugin_findologicxmlexport();
+        $adminPlugin->html();
+        $output = ob_get_clean();
+        $dom = new \DOMDocument('1.0', 'UTF-8');
+
+
+        $internalErrors = libxml_use_internal_errors(true);
+        $dom->loadHTML($output);
+        libxml_use_internal_errors($internalErrors);
+
+        $header = $dom->getElementsByTagName('h1')[0];
+        $header = $header->nodeValue;
+
+        $expectedHeader = 'FINDOLOGIC XML Export Plugin';
+
+        $legend = $dom->getElementsByTagName('legend')[0];
+        $legend = $legend->nodeValue;
+
+        $expectedLegend = 'Pages without title (2)';
+
+        $this->assertEquals($expectedHeader, $header, 'Expected header should be equal to the translation set in the english translation file.');
+        $this->assertEquals($expectedLegend, $legend, 'Expected legend should be equal to the translation set in the english translation file and should contain one item when one page has no title set.');
+    }
+
+    public function test_response_have_multible_elements_when_six_or_more_pages_has_no_title_set()
+    {
+        $pageHasNoTitle = ['noootitle0', 'noootitle1', 'noootitle2', 'noootitle3', 'noootitle4', 'noootitle5'];
+        Helper::savePages($pageHasNoTitle);
+
+        ob_start();
+
+        $adminPlugin = new admin_plugin_findologicxmlexport();
+        $adminPlugin->html();
+        $output = ob_get_clean();
+        $dom = new \DOMDocument('1.0', 'UTF-8');
+
+        $internalErrors = libxml_use_internal_errors(true);
+        $dom->loadHTML($output);
+        libxml_use_internal_errors($internalErrors);
+
+        $finder = new DomXPath($dom);
+
+        $pageFields = ['page-id', 'page-url', 'page-author', 'page-last-edited'];
+
+        foreach ($pageFields as $key => $pageField) {
+
+            $nodes = $finder->query("//td[contains(@class, '$pageField')]");
+
+            for ($i = 0; $i < 5; $i++) {
+                $pageWithoutTitle[$pageField][] = trim($nodes->item($i)->nodeValue);
+            }
+        }
+
+        // page-id
+        $expectedPageWithoutTitle['page-id'][0] = 'noootitle0';
```

Comment:
1. Build an array with expected page titles
2. Build an array with actual pages titles
3. Sort both by the same criteria (alpha most likely)
4. Compare them with the appropriate assert 

See https://stackoverflow.com/a/12496979/1478499

Reply:
Note4Me: should be done locally